### PR TITLE
fix: ignore broken symlinks

### PR DESCRIPTION
**Pull request by vdayanand**
Currently formatting breaks if it encounters broken symlink in the directory

### CODE REVIEW

**Review by domluna — APPROVED** [review]

**Comment by vdayanand** [comment]
Looks like the CI failure is unrelated?

**Comment by pfitzseb** [comment]
Yeah, the Julia 1.4 distribution for MacOS is broken. I'd suggest adding a [exclusion](https://github.com/julia-vscode/CSTParser.jl/blob/3ea92f4e40052d87d52aedabb16b550b5ac539ec/.github/workflows/jlpkgbutler-ci-pr-workflow.yml#L14-L16):
```
        exclude:
          - os: macos-13
          - julia-version: 1.4
```
Also probably a good idea to switch back to `macos-13` for now, `macos-latest` got switched to ARM machines or something recently.

**Comment by vdayanand** [comment]
> Also probably a good idea to switch back to macos-13 for now, macos-latest got switched to ARM machines or something recently.

Did you mean switching to macos-latest (which is currently excluded) instead of macos-13?

**Comment by pfitzseb** [comment]
I meant using `macos-13` globally and excluding Julia 1.4. I've also been using the juliaup action recently -- here's how that would look like
```
...
    strategy:
      matrix:
        julia-version: ['1.0', '1.1', '1.2', '1.3', '1.4', '1.5', '1.6', '1.7', '1.8', '1.9', '1.10', '1.11']
        os: [ubuntu-latest, windows-latest, macos-13]
        exclude:
          - os: macos-13
          - julia-version: 1.4
    steps:
      - uses: actions/checkout@v4
      - uses: julia-actions/install-juliaup@v1
        with:
           julia-version: ${{matrix.julia-version}}
...
```

**Comment by vdayanand** [comment]
Thanks. I have updated the CI

**Comment by domluna** [comment]
why isnt the ci running hmmm

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -28,10 +28,10 @@ jobs:
           - '1.10'
           # - 'nightly'
         arch: [x64, x86]
-        os: [ubuntu-latest, windows-latest, macos-latest]
+        os: [ubuntu-latest, windows-latest, macos-13]
         exclude:
-          - os: macos-latest
-            arch: x86
+          - os: macos-13
+          - julia-version: 1.4
```
Comment (pfitzseb) [inline .github/workflows/ci.yml:34]: Needs to be 
```suggestion
            version: 1.4
```
I think.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -28,10 +28,10 @@ jobs:
           - '1.10'
           # - 'nightly'
         arch: [x64, x86]
-        os: [ubuntu-latest, windows-latest, macos-latest]
+        os: [ubuntu-latest, windows-latest, macos-13]
         exclude:
-          - os: macos-latest
-            arch: x86
+          - os: macos-13
+            version: 1.4
```
Comment (kimikage) [inline .github/workflows/ci.yml:34]: The original exclusion here is `arch: x86`. Why change it?

Comment (kimikage) [reply .github/workflows/ci.yml:34]: see PR #847

**Pull request merged**